### PR TITLE
[Merged by Bors] - load zeroed UVs as fallback in gltf loader

### PR DESCRIPTION
**Pull request by jakobhellermann**
fixes a lot of gltf loading failures (see https://github.com/bevyengine/bevy/issues/1802)

### CODE REVIEW

**Comment by mockersf** [comment]
a message should be logged to notify the user of the change, and let them know if they want to fix it in their modelling tool

**Comment by jakobhellermann** [comment]
> a message should be logged to notify the user of the change, and let them know if they want to fix it in their modelling tool

I added a log.

File: crates/bevy_gltf/src/loader.rs [hunk 1]
```diff
@@ -140,6 +140,11 @@ async fn load_gltf<'a, 'b>(
                 .map(|v| VertexAttributeValues::Float2(v.into_f32().collect()))
             {
                 mesh.set_attribute(Mesh::ATTRIBUTE_UV_0, vertex_attribute);
+            } else {
+                let len = mesh.count_vertices();
+                let uvs = vec![[0.0, 0.0]; len];
+                bevy_log::warn!("missing `TEXCOORD_0` vertex attribute, loading zeroed out UVs");
```
Comment (cart) [inline crates/bevy_gltf/src/loader.rs:146]: Warnings (and to a lesser extent, info logs) in Bevy _must_ be actionable/fixable. But there will be cases where users won't be able to modify models (ex: they bought them on an asset store). I think we should consider "uv-less" models a "happy path" case.  

Can we make this a `debug` log instead?

Comment (jakobhellermann) [reply crates/bevy_gltf/src/loader.rs]: Done.

**Comment by cart** [comment]
I'm going to hold off on merging this until we have this conversation: https://github.com/bevyengine/bevy/pull/1808#discussion_r612720797

We should sort out a short term and long term plan for "missing vertex attributes". Long term I'd like to support loading "minimal" meshes without generating vertex attributes when they aren't needed, or warning/logging unnecessarily.

**Comment by cart** [comment]
Short term these changes are super valuable / make gltf loading more usable, so I'm willing to make concessions. I just want to be thoughtful about our first steps.

**Comment by jakobhellermann** [comment]
The two possible solutions are 
1. add zeroed buffer for UVs
2. handle missing UVs in the shader (like #1010)

In #1010 it seemed like you preferred zeroed buffers as the solution.

Am I missing any alternatives?

**Comment by cart** [comment]
(sorry for the delay)

I think generating UVs is probably the right call, but as I [called out here](https://github.com/bevyengine/bevy/pull/1808#discussion_r612720797), I think long term this should probably be a "mesh transform" users opt in to rather than something that happens implicitly. I expect there to be a class of meshes that don't need certain vertex attributes. Rather than bloat them with those attributes, I'd rather detect cases of "shader expected attributes that the mesh doesn't have", then enable the user to opt-in to generating things like "zeroed out uvs". 

Currently we don't support "asset config", so theres no good way to opt in to these transformations. For now, I'm down to merge this.

**Pull request closed without merging**